File: LP3/DAA/fibonaaci/Fibonacci.java
package Java;

public class Fibonacci {

    public static void main(String[] args) {
        int n = 10; // Specify the position in the Fibonacci sequence
        int result = fibonacci(n);
        System.out.println("Fibonacci number at position " + n + " is: " + result);
    }

    public static int fibonacci(int n) {
        if (n <= 1) {
            return n;
        }

        int a = 0, b = 1, c = 0;
        int stepCount = 0;

        for (int i = 2; i <= n; i++) {
            c = a + b;
            a = b;
            b = c;
            stepCount++;
        }

        System.out.println("Step count: " + stepCount);
        return c;
    }
    
}

File: LP3/DAA/fibonaaci/fibonacci.py
def fibonacci_recursive(n, step_count=[0]):
    step_count[0] += 1
    if n <= 1:
        return n
    return fibonacci_recursive(n-1, step_count) + fibonacci_recursive(n-2, step_count)

def fibonacci_iterative(n):
    step_count = 0
    a = 0
    b = 1
    for _ in range(n):
        step_count += 1
        temp = a + b
        a = b
        b = temp
    return a, step_count

n = int(input("Enter the term number: "))

recursive_steps = [0]
fib_recursive = fibonacci_recursive(n, recursive_steps)
print(f"Recursive: Fibonacci({n}) = {fib_recursive}, Steps = {recursive_steps[0]}")

fib_iterative, iterative_steps = fibonacci_iterative(n)
print(f"Iterative: Fibonacci({n}) = {fib_iterative}, Steps = {iterative_steps}")
